virocon/distributions.py

```diff
@@ -173,10 +354,41 @@ def _get_rvs_size(n, pars):
             return n
 
 
-
 class WeibullDistribution(Distribution):
+    """
+     A weibull distribution.
+
+    :math:`f(x) = \\frac{k}{\\lambda}\\left( \\frac{x-\\theta}{\\lambda}\\right)^{k -1} \\exp \\left[-\\left( \\frac{x-\\theta}{\\lambda} \\right)^{k} \\right]`
     
+    Parameters
+    ----------
+    lambda : float
+        Scale parameter of the weibull distribution. Defaults to 1.
+    k : float
+        Shape parameter of the weibull distribution. Defaults to 1.
+    theta : float
+        Location parameter of the weibull distribution (3-parameter weibull
+        distribution). Defaults to 0.
+    f_lambda : float
+        Fixed scale parameter of the weibull distribution (e.g. given physical
+        parameter). If this parameter is set, lambda is ignored. Defaults to 
+        None.
+    f_k : float
+       Fixed shape parameter of the weibull distribution (e.g. given physical
+       parameter). If this parameter is set, k is ignored. Defaults to 
+       None. 
+    f_theta : float
+        Fixed location parameter of the weibull distribution (e.g. given physical
+        parameter). If this parameter is set, theta is ignored. Defaults to 
+        None.
+    fit_method : float
+        Method of estimating the parameters of a probability distribution. 
+        Defaults to maximum likelihood estimation (mle).
+    weights : KAI???
```

Comment:
Weights for weighted-least-squares fitting. Ignored (not implemented) for WeibullDistribution.